timefold-solver-python-core/src/main/python/score/_score_analysis.py

```diff
@@ -541,6 +597,73 @@ def constraint_analyses(self) -> list[ConstraintAnalysis]:
                 list['_JavaConstraintAnalysis[Score]'], self._delegate.constraintAnalyses())
         ]
 
+    @overload
+    def constraint_analysis(self, constraint_package: str, constraint_name: str) -> ConstraintAnalysis:
+        ...
+
+    @overload
+    def constraint_analysis(self, constraint_ref: 'ConstraintRef') -> ConstraintAnalysis:
+        ...
+
+    def constraint_analysis(self, *args) -> ConstraintAnalysis:
+        """
+        Performs a lookup on `constraint_map`.
```

Comment:
This docstring need to be refactored. Its signature is not
`(constraint_package : str
   constraint_name : str
   constraint_ref : ConstraintRef) -> ConstraintAnalysis`
   
  but rather
  
  `(constraint_package : str
   constraint_name : str) -> ConstraintAnalysis |
   (constraint_ref : ConstraintRef) -> ConstraintAnalysis`
   
   This should do:

```python
"""
Performs a lookup on `constraint_map`.


Parameters
----------
*args: *tuple[str, str] | *tuple[ConstraintRef]
    Either two strings or a single ConstraintRef can be passed as positional arguments.
    If two strings are passed, they are taken to be the constraint package and constraint name respectively.
    If a ConstraintRef is passed, it is used to perform the lookup.

Returns
-------
ConstraintAnalysis
    None if no constraint matches of such constraint are present
"""
```